lightning/src/util/sweep.rs

```diff
@@ -0,0 +1,675 @@
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! This module contains an [`OutputSweeper`] utility that keeps track of
+//! [`SpendableOutputDescriptor`]s, i.e., persists them in a given [`KVStore`] and regularly retries
+//! sweeping them.
+
+use crate::chain::chaininterface::BroadcasterInterface;
+use crate::chain::channelmonitor::ANTI_REORG_DELAY;
+use crate::chain::{self, BestBlock, Confirm, Filter, Listen, WatchedOutput};
+use crate::{impl_writeable_tlv_based, log_error, log_debug};
+use crate::io;
+use crate::ln::ChannelId;
+use crate::prelude::{String, Vec, Box};
+use crate::sign::{EntropySource, SpendableOutputDescriptor};
+use crate::sync::Mutex;
+use crate::util::persist::{KVStore, SPENDABLE_OUTPUT_INFO_PERSISTENCE_PRIMARY_NAMESPACE,
+	SPENDABLE_OUTPUT_INFO_PERSISTENCE_SECONDARY_NAMESPACE};
+use crate::util::ser::{Readable, Writeable};
+use crate::util::logger::Logger;
+
+use bitcoin::blockdata::block::Header;
+use bitcoin::{BlockHash, Transaction, Txid};
+
+use core::ops::Deref;
+use core::fmt::Write;
+use core::cmp;
+
+/// The default interval after which we regenerate output spending transactions.
+pub const DEFAULT_REGENERATE_SPEND_THRESHOLD: u32 = 144;
```

Comment:
nit: I had to read in the code below in order to get the unit(blocks?) of this interval, it would be useful to add it to the docs